MatrixSDK/Data/MXRoomSummaryUpdater.m

```diff
@@ -113,8 +113,15 @@ - (BOOL)session:(MXSession *)session updateRoomSummary:(MXRoomSummary *)summary
     // Accept event which type is in the filter list
     if (event.eventId && (_lastMessageEventTypesAllowList == nil || [_lastMessageEventTypesAllowList containsObject:event.type]))
     {
-        [summary updateLastMessage:[[MXRoomLastMessage alloc] initWithEvent:event]];
-        updated = YES;
+        // Don't accept event related to profile change
+        // TODO: Add a flag if needed to configure this
+        // TODO: Only accept membership changes from current user
+        if (event.eventType == MXEventTypeRoomMember 
+            && !event.isUserProfileChange)
+        {
+            [summary updateLastMessage:[[MXRoomLastMessage alloc] initWithEvent:event]];
+            updated = YES;        
+        }
```

Comment:
This currently prevents the last message from updating for the allowed types (such as messages). The logic should probably be flipped like so:
```suggestion
        if (event.eventType == MXEventTypeRoomMember && event.isUserProfileChange)
        {
            return NO;
        }
        
        [summary updateLastMessage:[[MXRoomLastMessage alloc] initWithEvent:event]];
        updated = YES;
```